framework/web/User.php

```diff
@@ -559,9 +593,13 @@ protected function sendIdentityCookie($identity, $duration)
      * If null, it means switching the current user to be a guest.
      * @param integer $duration number of seconds that the user can remain in logged-in status.
      * This parameter is used only when `$identity` is not null.
+     * @param boolean $cookieBased whether the login is cookie-based
      */
-    public function switchIdentity($identity, $duration = 0)
+    public function switchIdentity($identity, $duration = 0, $cookieBased = false)
```

Comment:
Hi,  sorry for the mess.  I think I tried to do too much at the same time.  I created a new pull request just moving the cookie code into two new protected functions.